File: 0257-binary-tree-paths/0257-binary-tree-paths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        res = []

        def dfs(node, res, st):
            if not node:
                return 
            st += str(node.val) + "->"
            if not node.left and not node.right:
                res.append(st[:-2])
            else:
                dfs(node.left, res, st)
                dfs(node.right, res, st)
        
        dfs(root, res, "")
        return res

File: 0257-binary-tree-paths/0257-binary-tree-paths.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function binaryTreePaths(root: TreeNode | null): string[] {
    let st: [TreeNode, string][] = [];
    let res: string[] = [];

    st.push([root, "" + root.val]);

    while(st.length > 0) {
        let [node, path] = st.pop();
        
        if(node.left === null && node.right === null) {
            res.push(path);
        } else {
            if(node.left !== null)
                st.push([node.left, path+"->"+node.left.val]);
            if(node.right !== null)
                st.push([node.right, path+"->"+node.right.val]);
        }
    }
    return res;
};